monitoring/monitoring-plugins/neteye/check_ingress_elasticsearch_index.sh

```diff
@@ -0,0 +1,264 @@
+  echo -e "\t--index-static-name (str): the static name of the index (default: ${INDEX_STATIC_NAME})"
+  echo -e "\t--ingested-time-field (str): the name of the field to use (default: ${INGESTED_TIME_FIELD})"
+  echo -e "\t--index-creation-interval (str): it specifies whenever a new index is created (e.g., d = daily, M = monthly) (default ${INDEX_CREATION_INTERVAL})"
+  echo -e "\t--es-host (str): the elasticsearch host or ip (default: ${ES_HOST})"
+  echo -e "\t--es-port (int): the elasticsearch port (default: ${ES_PORT})"
+  echo -e "\t--es-protocol (str): the protocol used to connect to elasticsearch (default: ${ES_PROTOCOL})"
+  echo -e "\t--output-date-format (str): the output date format compatible with 'date' command (default: '${OUTPUT_DATE_FORMAT}')"
+  echo -e "\t--curl-command-path (str): path to a curl executable to use (default: ${CURL_COMMAND_PATH}"
+  echo -e "\t--curl-cert (str): path to the client's certificate (check: man curl for details) (default: ${CURL_CERT}"
+  echo -e "\t--curl-key (str): path to the private key of the client (check: man curl for details) (default: ${CURL_PRIVATE_KEY}"
+}
+
+
+# Retrieving the command line options. Each shift, discard one argument.
+while [[ $# -gt 0 ]]
+do
+  key="$1"
+
+  case ${key} in
+      "-c"|"--critical-threshold")
+      CRITICAL_THRESHOLD="$2"
+      shift
+      shift
+      ;;
+      "-w"|"--warning-threshold")
+      WARNING_THRESHOLD="$2"
+      shift
+      shift
+      ;;
+      "--threshold-format")
+      THRESHOLD_FORMAT="$2"
+      shift
+      shift
+      ;;
+      "--index-date-format")
+      INDEX_DATE_FORMAT="$2"
+      shift
+      shift
+      ;;
+      "--index-static-name")
+      INDEX_STATIC_NAME="$2"
+      shift
+      shift
+      ;;
+      "--ingested-time-field")
+      INGESTED_TIME_FIELD="$2"
+      shift
+      shift
+      ;;
+      "--index-creation-interval")
+      INDEX_CREATION_INTERVAL="$2" # d = daily, m = monthly, y = yearly
+      shift
+      shift
+      ;;
+      "--es-host")
+      ES_HOST="$2"
+      shift
+      shift
+      ;;
+      "--es-port")
+      ES_PORT="$2"
+      shift
+      shift
+      ;;
+      "--es-protocol")
+      ES_PROTOCOL="$2"
+      shift
+      shift
+      ;;
+      "-h"|"--help")
+      help "${PROGRAM_NAME}"
+      exit 2
+      ;;
+      "--curl-command-path")
+      CURL_COMMAND_PATH="$2"
+      shift
+      shift
+      ;;
+      "--curl-cert")
+      CURL_CERT="$2"
+      shift
+      shift
+      ;;
+      "--curl-key")
+      CURL_PRIVATE_KEY="$2"
+      shift
+      shift
+      ;;
+      *)    # unknown option
+        echo "Unknown parameter $key: $2"
+        help
+        exit 3
+      ;;
+  esac
+done
+
+
+
+#
+# Translate threshold from ${THRESHOLD_FORMAT} to seconds to allow flexibility
+# in the definition
+#
+THRESHOLD_STRING="seconds"
+
+case ${THRESHOLD_FORMAT} in
+  "d"|"days")
+    CRITICAL_THRESHOLD=$((CRITICAL_THRESHOLD*86400))
+    WARNING_THRESHOLD=$((WARNING_THRESHOLD*86400))
```

Comment:
Should THRESHOLD_FORMAT and THRESHOLD_STRING not be related to each other? In my opinion I would a) just use  the THRESHOLD_FORMAT variable and b) have define a default value for it

Reply:
Actually, THRESHOLD_FORMAT is the format used for input. E.g., a user can specify a THRESHOLD in days, hours, minutes and seconds, but internally we use only seconds for calculation. Therefore, I propose the occurrences of "THRESHOLD_STRING" with its value "seconds".